### ISSUE TITLE
Safari fails on crossdomain image

### ISSUE THREAD

**Issue by yuvadm** [paperjs/paper.js#711]
My paper.js application takes a remote image:

```html
<img src="https://remote.example.com/foo.jpg" crossdomain="anonymous">
```

and applies some raster processing on it (javascript, not paperscript):

```js
var canvas = document.getElementById('myCanvas');
paper.setup(canvas);
var raster = new paper.Raster('myImg');
doStuff(raster);
```

Remote domain is adding proper CORS headers (namely, `Access-Control-Allow-Origin=*`) and indeed this works as expected on Firefox and Chrome. Safari on both OS X as well as iOS 8.3 fails on this with:

```
Unable to get image data from canvas because the canvas has been tainted by cross-origin data.
[Error] SecurityError: DOM Exception 18: An attempt was made to break through the security policy of the user agent.
```

This might or might not be a paper.js bug, but I'm curious if anyone knows why Safari isn't honoring the cross-domain policy.

**Comment by lehni** [comment]
I'm pretty sure that's a browser issue, not a Paper.js issue. You could try loading the raster directly from the URL instead of through the <img> tag:

```js
var raster = new paper.Raster('https://remote.example.com/foo.jpg');
raster.onLoad = function() {
    doStuff(raster);
}
```

**Comment by lehni** [comment]
Or even better:

```js
var raster = new Raster({
    source: 'http://assets.paperjs.org/images/marilyn.jpg',
    onLoad: function() {
        doStuff(this);
    }
});
```

**Comment by yuvadm** [comment]
No dice:

```js
> var raster = new paper.Raster('https://example.com/foo.jpg')
< undefined
> raster.getImageData()
[Error] Unable to get image data from canvas because the canvas has been tainted by cross-origin data.
Error: SecurityError: DOM Exception 18
```

**Comment by lehni** [comment]
Well like I said, it's not a paper.js issue. But maybe you can provide the actual URL so I can look into it.

**Comment by yuvadm** [comment]
@lehni Any image from the Instagram CDN. e.g. https://scontent.cdninstagram.com/hphotos-xaf1/t51.2885-15/s150x150/e15/11312322_667939720005261_1131816232_n.jpg

**Comment by lehni** [comment]
I get the same error in Chrome:

http://sketch.paperjs.org/#S/LU89a8MwFPwrQktsSCRLRo6j0K1LoVM7NsW8yortBj8Z66UNBP/3yqQHN9zHcHfnCKPnlr9fPLmeb7kL7arR/7I3iOTn7H5ClhDDdXbesk1PNEUrZXQBySMJ1+KAkaCbYRQujLKf+kAh7m5wVpKMErquzU4ZGZUpbonSJ6FUqXSpdVNV+0N52OuiKIyuVLMGtapS1qD4nrrN9rEg4GuA1rLzFR0NAbOc/W9b8QMza4GAPTHqhyg6Ty8jdP45eVl+fBSXEy75Md38mj1cpjAgRW4/Ppc/

**Comment by yuvadm** [comment]
@lehni Then your method isn't good. You must use an `img` tag with `crossdomain="anonymous"`. That works perfectly on FF and Chrome.

**Comment by lehni** [comment]
This has been fixed in #754
